src/main/scala/scorex/core/validation/ModifierValidator.scala

```diff
@@ -119,9 +119,9 @@ case class ValidationState[T](result: ValidationResult[T], strategy: ValidationS
 
   /** Validate the `id`s are equal. The `error` callback will be provided with detail on argument values
     */
-  def validateEqualIds(given: => Array[Byte], expected: => Array[Byte])
+  def validateEqualIds(given: => ModifierId, expected: => ModifierId)
                       (error: String => Invalid): ValidationState[T] = {
-    validate(java.util.Arrays.equals(given, expected)) {
+    validate(given == expected) {
       error(s"Given: ${e.encode(given)}, expected ${e.encode(expected)}")
```

Comment:
we can drop now `e.encode` from here and get rid of `implicit encoder: ScorexEncoder` in `ModifierValidator`